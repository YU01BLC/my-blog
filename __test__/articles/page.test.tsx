import { render, screen } from "@testing-library/react";
import Page from "../../src/app/articles/page";
import "@testing-library/jest-dom";

describe("PageコンポーネントSnapshot", () => {
  it("snapshot", () => {
    const { asFragment } = render(<Page />);
    expect(asFragment()).toMatchSnapshot();
  });
});
describe("Pageコンポーネント表示確認", () => {
  it("テキスト「articles」が表示されること", () => {
    render(<Page />);
    const headElement = screen.getByText("articles");
    expect(headElement).toBeInTheDocument();
  });
});
