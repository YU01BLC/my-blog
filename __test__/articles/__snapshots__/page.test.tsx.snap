// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page(Article)コンポーネントSnapshot snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-i3n67o"
    data-testid="article"
  >
    <div
      class="MuiBox-root css-6gs9hh"
    >
      <div
        class="MuiBox-root css-0"
        data-testid="article-list"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
              >
                記事タイトル
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
              >
                2023/12/1
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <a
              href="/hoge"
            >
              <img
                alt="Paella dish"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                height="100%"
                src="https://source.unsplash.com/collection/1346951/1000x500?sig=2"
                width="100%"
              />
            </a>
            <p
              class="MuiTypography-root MuiTypography-body1 css-136xlp0-MuiTypography-root"
              data-testid="article-list--text"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Veritatis quod maiores totam aperiam necessitatibus est, exercitationem architecto quibusdam accusamus maxime asperiores eaque adipisci itaque nemo, minus odio, alias dolor possimus!
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        data-testid="dummy-article"
      >
        <div
          class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              記事タイトル(dummy)
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
            >
              2023/12/1
            </span>
          </div>
        </div>
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <img
            alt="Paella dish"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
            height="194"
            src="/assets/dummy-blog-image.jpg"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 css-136xlp0-MuiTypography-root"
          >
            ダミー記事です。
            <br />
            右上の「記事を書く」ボタンから新規記事を執筆してください。
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-136xlp0-MuiTypography-root"
            data-testid="dummy-article--text"
          >
            ※   記事投稿後ダミー記事は非表示となります。
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
